lib/nimble_totp.ex

```diff
@@ -120,7 +120,7 @@ defmodule NimbleTOTP do
 
   """
   @spec otpauth_uri(String.t(), String.t(), keyword()) :: String.t()
-  def otpauth_uri(label, secret, uri_params \\ []) do
+  def otpauth_uri(label, secret, uri_params \\ []) when is_binary(label) and is_binary(secret) do
```

Comment:
The guards here and below all follow the specs, so I think this is fine. I didn't add them to `valid?` as that seems to explicitly return `false` on every invalid input, regardless of what the spec says.